jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -156,36 +156,39 @@ public StringBuffer() {
  * @param		capacity	the initial capacity
  */
 public StringBuffer(int capacity) {
-	if (String.enableCompression) {
-		/*[IF Sidecar19-SE]*/
-		if (capacity <= MAX_CAPACITY) {
-			value = new byte[capacity];
-		} else {
-			/*[MSG "K05df", "Unable to allocate an array of the specified capacity. The maximum supported capacity is Integer.MAX_VALUE / 2."]*/
-			throw new OutOfMemoryError(com.ibm.oti.util.Msg.getString("K05df")); //$NON-NLS-1$
-		}
-		/*[ELSE]*/
-		if (capacity == Integer.MAX_VALUE) {
-			value = new char[(capacity / 2) + 1];
-		} else {
-			value = new char[(capacity + 1) / 2];
+	/* capacity argument is modified before the byte/char array is created. If
+	 * capacity argument is Integer.MIN_VALUE (-2147483648), then capacity *= 2
+	 * will yield a non-negative number due to overflow. We will fail to throw
+	 * NegativeArraySizeException. The check below will assure that
+	 * NegativeArraySizeException is thrown if capacity argument is less than 0.
+	 */
+	if (capacity < 0) {
+		throw new NegativeArraySizeException();
+	}
+	
+	/*[IF !Sidecar19-SE]*/
+	this.capacity = capacity;
```

Comment:
This should be at the end of the method as we want the allocation of the array to occur before the capacity is assigned.

Reply:
fixed.